tests/llm/utils/mock_toolset.py

```
@@ -29,53 +36,69 @@
+def ensure_non_error_returned(
+    original_tool_result: Optional[StructuredToolResult],
+) -> StructuredToolResult:
+    if original_tool_result and original_tool_result.status in [
+        ToolResultStatus.SUCCESS,
+        ToolResultStatus.NO_DATA,
+    ]:
+        return original_tool_result
+    else:
+        return StructuredToolResult(
+            status=ToolResultStatus.NO_DATA, params=original_tool_result.params
+        )
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)